sdks/go/pkg/beam/core/runtime/exec/sdf_invokers_test.go

```diff
@@ -0,0 +1,262 @@
+					t.Fatalf("invokeSplitRestriction failed: %v", err)
+				}
+				if !cmp.Equal(got, test.want) {
+					t.Errorf("invokeSplitRestriction(%v, %v) has incorrect output: got: %v, want: %v",
+						test.elms, test.rest, got, test.want)
+				}
+			})
+		}
+	})
+
+	t.Run("invokeRestrictionSize", func(t *testing.T) {
+		tests := []struct {
+			name string
+			sdf  *graph.SplittableDoFn
+			elms *FullValue
+			rest Restriction
+			want float64
+		}{
+			{
+				"SingleElem",
+				sdf,
+				&FullValue{Elm: 5},
+				Restriction{3},
+				8,
+			}, {
+				"KvElem",
+				kvsdf,
+				&FullValue{Elm: 5, Elm2: 2},
+				Restriction{3},
+				10,
+			},
+		}
+		for _, test := range tests {
+			test := test
+			fn := test.sdf.RestrictionSizeFn()
+			t.Run(test.name, func(t *testing.T) {
+				got, err := invokeRestrictionSize(fn, test.elms, test.rest)
+				if err != nil {
+					t.Fatalf("invokeRestrictionSize failed: %v", err)
+				}
+				if !cmp.Equal(got, test.want) {
+					t.Errorf("invokeRestrictionSize(%v, %v) has incorrect output: got: %v, want: %v",
+						test.elms, test.rest, got, test.want)
+				}
+			})
+		}
+	})
+
+	t.Run("invokeCreateTracker", func(t *testing.T) {
+		tests := []struct {
+			name string
+			sdf  *graph.SplittableDoFn
+			rest Restriction
+			want *RTracker
+		}{
+			{
+				"SingleElem",
+				sdf,
+				Restriction{3},
+				&RTracker{
+					Restriction{3},
+					1,
+				},
+			}, {
+				"KvElem",
+				kvsdf,
+				Restriction{5},
+				&RTracker{
+					Restriction{5},
+					2,
+				},
+			},
+		}
+		for _, test := range tests {
+			test := test
+			fn := test.sdf.CreateTrackerFn()
+			t.Run(test.name, func(t *testing.T) {
+				got, err := invokeCreateTracker(fn, test.rest)
+				if err != nil {
+					t.Fatalf("invokeCreateTracker failed: %v", err)
+				}
+				if !cmp.Equal(got, test.want) {
+					t.Errorf("invokeCreateTracker(%v) has incorrect output: got: %v, want: %v",
+						test.rest, got, test.want)
+				}
+			})
+		}
+	})
+}
+
+type Restriction struct {
+	Val int
+}
+
+// RTracker's methods can all be no-ops, we just need it to implement sdf.RTracker.
+type RTracker struct {
+	Rest Restriction
+	Val  int
+}
+
+func (rt *RTracker) TryClaim(interface{}) bool                      { return false }
+func (rt *RTracker) GetError() error                                { return nil }
+func (rt *RTracker) TrySplit(fraction float64) (interface{}, error) { return nil, nil }
+func (rt *RTracker) GetProgress() float64                           { return 0 }
+func (rt *RTracker) IsDone() bool                                   { return false }
+
+// In order to test that these methods get called properly, each one has an
+// implementation that lets us confirm that each argument was passed properly.
+
+type Sdf struct {
+}
+
+// Creates a restriction with the given value.
+func (fn *Sdf) CreateInitialRestriction(i int) Restriction {
+	return Restriction{i}
+}
+
+// Outputs two restrictions, the first containing the sum of i and rest.Val, the
```

Comment:
While this type and methods are only for test purposes, please adhere to the Go documentation idiom of having the method name in full to start the comment.